bindings/bs-promise/src/bs_promise.mli

```diff
@@ -27,37 +27,28 @@
    'rej - type the promise will be rejected with *)
 type ('res, 'rej) t
 
+external make : (('res -> unit) -> ('rej -> unit) -> unit) -> ('res, 'rej) t = "Promise" [@@bs.new]
+external create : (('res -> unit [@bs]) -> ('rej -> unit [@bs]) -> unit [@bs]) -> ('res, 'rej) t = "Promise" [@@bs.new]
+[@@ocaml.deprecated "Please use `make` instead"]
 
-(* then *)
-external then_ : ('res, 'rej) t -> ('res -> ('a, 'b) t [@bs]) -> ('a, 'b) t = "then" [@@bs.send]
-external (>>=) : ('res, 'rej) t -> ('res -> ('a, 'b) t [@bs]) -> ('a, 'b) t = "then" [@@bs.send]
-
-external thenValue : ('res, 'rej) t -> ('res -> 'a [@bs]) -> ('a, 'b) t = "then" [@@bs.send]
-external (>>|): ('res, 'rej) t -> ('res -> 'a [@bs]) -> ('a, 'b) t = "then" [@@bs.send]
-
-external thenWithError : ('res, 'rej) t -> ('res -> 'a [@bs]) -> ('rej -> 'b [@bs]) -> ('a, 'b) t = "then" [@@bs.send]
-
-
-(* catch *)
-external catch :  ('res, 'rej) t -> ('rej -> 'a [@bs]) -> ('a, 'b) t = "catch" [@@bs.send]
-external (>>?) :  ('res, 'rej) t -> ('rej -> 'a [@bs]) -> ('a, 'b) t = "catch" [@@bs.send]
-
-
-(* Promise.resolve *)
 external resolve : 'res -> ('res, 'a) t = "Promise.resolve" [@@bs.val]
-
-
-(* Promise.reject *)
 external reject : 'rej -> ('a, 'rej) t = "Promise.reject" [@@bs.val]
-
-
-(* Promise.all *)
 external all : ('res, 'rej) t array -> ('res array, 'rej) t = "Promise.all" [@@bs.val]
-
-
-(* Promise.race *)
 external race : ('res, 'rej) t array -> ('res, 'rej) t = "Promise.race" [@@bs.val]
 
+external then_ : ('res -> 'a) -> ('a, 'b) t = "then" [@@bs.send.pipe: ('res, 'rej') t]
+external thenValue : ('res, 'rej) t -> ('res -> 'a [@bs]) -> ('a, 'b) t = "then" [@@bs.send]
+[@@ocaml.deprecated "Please use `then_` instead"]

```

Comment:
The `deprecated` attribute? Yeah, I've been using that left and right after learning about it. Freedom! :)